cmd/machines.go

```diff
@@ -43,11 +44,101 @@ func init() {
 					setUserEnv, checkRequiredFlags, createMachine,
 				),
 			},
+			{
+				Name:  "list",
+				Usage: "List machines for an organization",
+				Flags: []cli.Flag{
+					orgFlag("org to the machine belongs to", true),
+					teamFlag("the team the machine belongs to", false),
+					destroyedFlag(),
+				},
+				Action: chain(
+					ensureDaemon, ensureSession, loadDirPrefs, loadPrefDefaults,
+					setUserEnv, checkRequiredFlags, listMachinesCmd,
+				),
+			},
 		},
 	}
 	Cmds = append(Cmds, machines)
 }
 
+func listMachinesCmd(ctx *cli.Context) error {
+	cfg, err := config.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	client := api.NewClient(cfg)
+	c := context.Background()
+
+	args := ctx.Args()
+	if len(args) > 0 {
+		return errs.NewUsageExitError("Too many arguments supplied.", ctx)
+	}
+
+	// Look up the target org
+	org, err := client.Orgs.GetByName(c, ctx.String("org"))
+	if err != nil {
+		return errs.NewExitError(envListFailed)
+	}
+	if org == nil {
+		return errs.NewExitError("Org not found.")
+	}
+	orgID := org.ID
+
+	state := primitive.MachineActiveState
+	if ctx.Bool("destroyed") {
+		state = primitive.MachineDestroyedState
+	}
+
+	teams, err := client.Teams.List(c, org.ID, ctx.String("team"), "")
```

Comment:
We should probably filter by "machine team", so we only return teams that a machine can belong too (via the type property).